adobe/spectrum-css
# feat(preview): add figma support to storybook [CSS-284]

- [CSS-284](https://jira.corp.adobe.com/browse/CSS-284)

## Description

Add support for Figma design links in Spectrum CSS storybook. Note: this does not add the actual design links to the stories but sets up the infrastructure to do so.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [ ] Open the [storybook](https://pr-1680--spectrum-css.netlify.app/preview/?path=/story/components-action-bar--default) for the action bar component:

1. Click the Design tab at the end of the panel, you should be prompted to log in:
    <img width="1849" alt="Screenshot 2024-02-07 at 10 22 19 AM" src="https://github.com/adobe/spectrum-css/assets/1840295/e6fe9905-cd7e-4c94-9cea-37c09aaf8ab0">
2. Make sure you allow 3rd party cookies; without them, the login pop-up may not show up
    <img width="281" alt="Screenshot 2024-02-07 at 10 22 27 AM" src="https://github.com/adobe/spectrum-css/assets/1840295/859f7ea6-fd64-4ca1-98e9-d73dbee7c447">
3. Log in using your @adobe SSO or with the account that was granted access to the Spectrum Figma library for viewing
    <img width="915" alt="Screenshot 2024-02-07 at 10 22 54 AM" src="https://github.com/adobe/spectrum-css/assets/1840295/9ecdcc19-0cf6-45cf-b359-8fe3806c0bb1">
4. Presto!
    <img width="1620" alt="Screenshot 2024-02-07 at 10 17 03 AM" src="https://github.com/adobe/spectrum-css/assets/1840295/2be437d0-156a-4763-8465-1d1481b50b9e">


### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:


2. If components have been modified, VRTs have been run on this branch:


## To-do list
- [ ] If my change impacts documentation, I have updated the documentation accordingly. (@todo)
